### ISSUE TITLE
App crashed after upgraded to React Native 0.69.6/React 18.0.0/Expo SDK46

### ISSUE THREAD

**Issue by alickcai** [facebook/react-native#34998]
### New Version

0.69.6

### Old Version

0.68.2

### Build Target(s)

Expo terminal - android & ios phone

### Output of `react-native info`

package.jason after upgraded

  "dependencies": {
    "@react-native-async-storage/async-storage": "~1.17.10",
    "@react-native-community/slider": "4.2.3",
    "@react-navigation/native": "^6.0.13",
    "@react-navigation/native-stack": "^6.9.1",
    "@types/react-native": "~0.69.13",
    "axios": "^1.1.3",
    "cos-js-sdk-v5": "^1.4.6",
    "expo": "^46.0.16",
    "expo-av": "~12.0.4",
    "expo-camera": "~12.3.0",
    "expo-contacts": "~10.3.0",
    "expo-dev-client": "~1.3.1",
    "expo-device": "~4.3.0",
    "expo-file-system": "~14.1.0",
    "expo-image-manipulator": "~10.4.0",
    "expo-media-library": "~14.2.0",
    "expo-video-player": "^2.2.0",
    "expo-web-browser": "~11.0.0",
    "pinyin-pro": "^3.11.0",
    "react": "18.0.0",
    "react-dom": "18.0.0",
    "react-native": "0.69.6",
    "react-native-animatable": "^1.3.3",
    "react-native-error-boundary": "^1.1.16",
    "react-native-screens": "~3.15.0",
    "react-native-web": "~0.18.9"
  },
  "devDependencies": {
    "@babel/core": "^7.19.3"
  },

### Issue and Reproduction Steps

Using Expo Go terminal to connect local server, App crashed with following message:

 ERROR  TypeError: undefined is not an object (evaluating 'navigator.userAgent.match')
Error: ENOENT: no such file or directory, open 'D:\Sunflower\webpackUniversalModuleDefinition@http:\192.168.2.105:19000\node_modules%5Cexpo%5CAppEntry.bundle?platform=ios&dev=true&hot=false'
    at Object.openSync (node:fs:585:3)
    at Object.readFileSync (node:fs:453:35)
    at getCodeFrame (D:\Sunflower\node_modules\metro\src\Server.js:979:18)
    at Server._symbolicate (D:\Sunflower\node_modules\metro\src\Server.js:1052:22)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Server._processRequest (D:\Sunflower\node_modules\metro\src\Server.js:443:7) {
  errno: -4058,
  syscall: 'open',
  code: 'ENOENT',
  path: 'D:\\Sunflower\\webpackUniversalModuleDefinition@http:\\192.168.2.105:19000\\node_modules%5Cexpo%5CAppEntry.bundle?platform=ios&dev=true&hot=false'
}

**Comment by alickcai** [comment]
it's caused by package cos-js-sdk-v5, closed this issue

**Comment by RomanSytnyk** [comment]
> it's caused by package cos-js-sdk-v5, closed this issue

How did you find it out?

I don't have such a package, but I face this error.
I got if after upgrading from 0.68.4 to 0.70.3

```
Error: ENOENT: no such file or directory, open '/Users/roman/IdeaProjects/atlas/home-module/http:/localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=app.dev.client.client&modulesOnly=false&runModule=true'
    at Object.openSync (node:fs:599:3)
    at Object.readFileSync (node:fs:467:35)
    at getCodeFrame (/Users/roman/IdeaProjects/atlas/home-module/node_modules/metro/src/Server.js:1028:18)
    at Server._symbolicate (/Users/roman/IdeaProjects/atlas/home-module/node_modules/metro/src/Server.js:1101:22)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Server._processRequest (/Users/roman/IdeaProjects/atlas/home-module/node_modules/metro/src/Server.js:458:7) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/Users/roman/IdeaProjects/atlas/home-module/http:/localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=app.dev.client.client&modulesOnly=false&runModule=true'
}
```

**Comment by alickcai** [comment]
I can explain more on my case:

I installed a 3rd-party package cos-js-sdk-v5 which worked well on version RN 0.68.2
I upgraded RN to 0.69.6 with cos-js-sdk-v5 upgraded automatically (per some dependency settings).

Unfortunately, the new version of cos-js-sdk-v5 didn't catch RN 0.69.6 so good (this package call some web function from a mobile instance), that's why I got above errors (...userAgent...).

I downgraded cos-js-sdk-v5 to old version and everything works.
This seems a typical error during upgrade with dependent packages not supporting each other, at least not on same pace.

Hope this helps, cheers.
